app/services/dashboard/data.rb

```diff
@@ -23,11 +23,14 @@ def latest_events
     con = ActiveRecord::Base.connection
     #workaround to get the inner query working
     ids = hosts.to_sql.sub(/SELECT.*FROM/, "SELECT #{Host.quoted_table_name}.#{con.quote_column_name('id')} FROM")
+    #sql optimization - instead of comparing dates, check just the id
+    oldest_report_id = ConfigReport.search_for('reported > "7 days ago"').last.try(:id)
+    oldest_report_id ||= ConfigReport.last.id + 1
     # 9 reports + header fits the events box nicely...
     @latest_events ||= ConfigReport.authorized(:view_config_reports).my_reports.interesting
                                    .where("#{ConfigReport.quoted_table_name}.#{con.quote_column_name('host_id')} IN (#{ids})")
-                                   .search_for('reported > "7 days ago"')
-                                   .limit(9).includes(:host)
+                                   .where("#{ConfigReport.quoted_table_name}.#{con.quote_column_name('id')} >= #{oldest_report_id}")
+                                   .limit(9).preload(:host)
```

Comment:
`includes` causes a left outer join. `preload` will cause an extra query to get the up to 9 relevant hosts which would be much faster then left joining two potentially huge tables.